### PR TITLE
Fix the dead loop in Mount.get_dir() in some cases

### PR DESCRIPTION
**Pull request by JoySnow**
Signed-off-by: XiaoXue Wang <xiaoxwan@redhat.com>

### CODE REVIEW

**Comment by JoySnow** [comment]
https://github.com/RedHatInsights/insights-core/issues/1245

**Comment by eduardocerqueira** [comment]
test this please

File: insights/parsers/tests/test_mount.py [hunk 1]
```diff
@@ -97,3 +97,20 @@ def test_mount():
     assert results.search(mount_options__contains='seclabel') == [
         results.rows[n] for n in (0, 1, 3, 4, 5, 7, 8, 11)
     ]
+
+
+MOUNT_WITHOUT_ROOT = """
```
Comment (xiangce) [inline insights/parsers/tests/test_mount.py]: This kind of mount content is not correct, for `mount` without the root '/', it's totally impossible and incorrect, for this cases, we should raise an Exception when finding this issue at the end of the parser.

**Comment by bfahr** [comment]
@JoySnow this change will break backward compatibility with existing plugins.  In this instance the change is to fix a bug which is an exception to the rule about breaking backward compatibility.  I just want to confirm that this is the correct approach.

**Comment by JoySnow** [comment]
@bfahr 
Those tests cases in existing plugins are not right without `/` mount point. (As we had supposed that  '/' must exists. ) 
I had fixed those test cases. We can merge this PR without any  incompatibility .

**Pull request merged**